modin/backends/base/query_compiler.py

```diff
@@ -1702,218 +3478,1392 @@ def set_index_names(self, names, axis=0):
+        return ResampleDefault.register(pandas.core.resample.Resampler.mean)(
+            self, resample_args, _method, *args, **kwargs
+        )
+
+    @_doc_resample_reduction(
+        result="median value", params="_method : str", link="median"
+    )
+    def resample_median(self, resample_args, _method, *args, **kwargs):
+        return ResampleDefault.register(pandas.core.resample.Resampler.median)(
+            self, resample_args, _method, *args, **kwargs
+        )
+
+    @_doc_resample_reduction(result="minimum value", params="_method : str", link="min")
+    def resample_min(self, resample_args, _method, *args, **kwargs):
+        return ResampleDefault.register(pandas.core.resample.Resampler.min)(
+            self, resample_args, _method, *args, **kwargs
+        )
+
+    @_doc_resample_fillna(method="'nearest'", link="nearest")
+    def resample_nearest(self, resample_args, limit):
+        return ResampleDefault.register(pandas.core.resample.Resampler.nearest)(
+            self, resample_args, limit
+        )
+
+    @_doc_resample_reduction(
+        result="number of unique values", params="_method : str", link="nunique"
+    )
+    def resample_nunique(self, resample_args, _method, *args, **kwargs):
+        return ResampleDefault.register(pandas.core.resample.Resampler.nunique)(
+            self, resample_args, _method, *args, **kwargs
+        )
+
+    # FIXME: Backend shouldn't care about differences between Series and DataFrame
+    # so `resample_ohlc_df` and `resample_ohlc_ser` should be combined.
+    @_doc_resample_agg(
+        action="compute open, high, low and close values",
+        params="_method : str",
+        output="labels of columns containing computed values",
+        link="ohlc",
+    )
+    def resample_ohlc_df(self, resample_args, _method, *args, **kwargs):
+        return ResampleDefault.register(pandas.core.resample.Resampler.ohlc)(
+            self, resample_args, _method, *args, **kwargs
+        )
+
+    @_doc_resample_agg(
+        action="compute open, high, low and close values",
+        params="_method : str",
+        output="labels of columns containing computed values",
+        link="ohlc",
+    )
+    def resample_ohlc_ser(self, resample_args, _method, *args, **kwargs):
+        return ResampleDefault.register(
+            pandas.core.resample.Resampler.ohlc, squeeze_self=True
+        )(self, resample_args, _method, *args, **kwargs)
+
+    @_doc_resample_fillna(method="'pad'", link="pad")
+    def resample_pad(self, resample_args, limit):
+        return ResampleDefault.register(pandas.core.resample.Resampler.pad)(
+            self, resample_args, limit
+        )
+
+    # FIXME: This method require us to build high-level resampler object
+    # which we shouldn't do at the backend. We need to move this at the front.
+    def resample_pipe(self, resample_args, func, *args, **kwargs):
+        """
+        Resample time-series data and apply aggregation on it.
```

Comment:
Need to add something about piping themself.

Reply:
What do you mean under "pipping themself"? Current doc-string represents what this function is: "apply passed function to the `pandas.Resampler` object".

Added reference to `Resampler.pipe` method.

Reply:
`pandas.Resampler.pipe` allows to call `pipe` multiple times for calls chaining. If there is no usage difference with pandas the reference will be enough.